providers/account.rb

```diff
@@ -210,3 +215,22 @@ def keygen_resource(exec_action)
     end
   end
 end
+
+def keypair_resource(exec_action)
+  new_resource.ssh_keypair.each do |name, key|
+    # avoid variable scoping issues in resource block
+    key_name, key_content = name, key
+
+    home = Etc.getpwnam(new_resource.username).dir
+    r = file "#{home}/.ssh/#{name}" do
+      content   key_content + "\n"
+      owner     new_resource.username
+      group     Etc.getpwnam(new_resource.username).gid
+      mode      '0600' unless key_name =~ /.pub$/
```

Comment:
It's an assumption, but one that is always true when using `ssh-keygen` unless you manually rename your private key to `*.pub`. :)

It will probably do no harm to just use 600 or even 400 for all files, the idea was to keep to the defaults (of ssh-keygen) if you can - And that would be 644 for the pubkey and 600 for the private one.